test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperExecutionContextTest.cs

```diff
@@ -153,14 +154,14 @@ public void HtmlAttributes_IgnoresCase(string originalName, string updatedName)
         {
             // Arrange
             var executionContext = new TagHelperExecutionContext("p", selfClosing: false);
-            executionContext.HTMLAttributes[originalName] = "hello";
+            executionContext.HTMLAttributes[originalName].Value = "hello";
```

Comment:
an implicit conversion from `string` in the `TagHelperAttribute` class would help tag helper authors avoid `Value` when the rest of the properties are uninteresting.  create `TagHelperAttribute` with `Name==null`.  set `Name` in the `TagHelperAttributes` setters, `Add()`, and `Insert()`.  `throw` if the method has no name to use but the property is `null`.